src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    @Test
+    void register_ageTooYoung_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(17);
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Користувач повинен бути не молодшим за 18 років", exception.getMessage());
+    }
+
+    @Test
+    void register_validUser_ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
+    void register_passwordEdgeCases_notOk() {
+        User user = new User();
+
+        user.setLogin("uniqueLogin1");
+        user.setPassword("");
+        user.setAge(20);
+        Exception exception1 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception1.getMessage());
+
+        user.setLogin("uniqueLogin2");
+        user.setPassword("abc");
+        Exception exception2 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception2.getMessage());
+
+        user.setLogin("uniqueLogin3");
+        user.setPassword("abcdf");
+        Exception exception3 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception3.getMessage());
+    }
+
+    @Test
+    void register_ageEdgeCases_notOk() {
+        User user = new User();
+        user.setPassword("validPassword");
+
+        user.setLogin("uniqueLogin4");
+        user.setAge(null);
+        Exception exception1 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Користувач повинен бути не молодшим за 18 років", exception1.getMessage());
+
+        user.setLogin("uniqueLogin5");
+        user.setAge(17);
+        Exception exception2 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Користувач повинен бути не молодшим за 18 років", exception2.getMessage());
+
+        user.setLogin("uniqueLogin6");
+        user.setAge(-1);
+        Exception exception3 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Користувач повинен бути не молодшим за 18 років", exception3.getMessage());
+    }
```

Comment:
Good job on testing the edge cases for age. However, you're missing the test case for a user that is exactly 18 years old. This is an important edge case to test.